humanmine.ipynb

```diff
@@ -0,0 +1,568 @@
+      "text/plain": [
+       "<BinaryConstraint: Gene.diseases.name CONTAINS diabetes>"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "query.add_constraint(\"proteinAtlasExpression.tissue.name\", \"=\", TISSUE)\n",
+    "query.add_constraint(\"proteinAtlasExpression.level\", \"ONE OF\", [\"Medium\", \"High\"])\n",
+    "query.add_constraint(\"organism.name\", \"=\", \"Homo sapiens\")\n",
+    "query.add_constraint(\"diseases.name\", \"CONTAINS\", DISEASE)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "proved-bronze",
+   "metadata": {},
+   "source": [
+    "Collect the genes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "adopted-treaty",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "upin_tissue = set()\n",
+    "for row in query.rows():\n",
+    "    upin_tissue.add(row[\"symbol\"])\n",
+    "genes = sorted(upin_tissue, reverse=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "selective-pathology",
+   "metadata": {},
+   "source": [
+    "Print out the list of genes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "secret-labor",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WFS1 VEGFA TCF7L2 TBC1D4 SOD2 SLC30A8 PTPN22 PDX1 \n",
+      "MIA3 KCNJ11 IRS2 IRS1 INSR INS IGF2BP2 IER3IP1 \n",
+      "HNF4A HNF1B HMGA1 HFE GPD2 GCK ENPP1 EIF2AK3 \n",
+      "DNAJC3 CEL CAPN10 APPL1 AKT2 ABCC8 "
+     ]
+    }
+   ],
+   "source": [
+    "for i, a in enumerate(genes):\n",
+    "    print(a, end=' ')\n",
+    "    if i % 8 == 7: \n",
+    "        print(\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "pending-fusion",
+   "metadata": {},
+   "source": [
+    "## Search for images in IDR associated to the genes found in Humanmine\n",
+    "\n",
+    "From the list of genes found using the intermine API, we are now looking in [Image Data Resource](https://idr.openmicroscopy.org/) for studies linked to those genes and with **TISSUE** as a ``Sample Type``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "following-blues",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TYPE = \"gene\"\n",
+    "SAMPLE_TYPE = \"tissue\"\n",
+    "EXPRESSION_KEY = \"Expression Pattern Description\"\n",
+    "EXPRESSION = \"Islets\" # \"Brain\"\n",
+    "STAGE = \"Developmental Stage\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "occupied-breakfast",
+   "metadata": {},
+   "source": [
+    "### Set up base URLS so can use shorter variable names later on"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "intermediate-belly",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "URL = \"https://idr.openmicroscopy.org/mapr/api/{key}/?value={value}&case_sensitive=false&orphaned=true\"\n",
```

Comment:
outwith this PR: would be nice if these weren't needed.

Reply:
agree